zerver/tests/test_subs.py

```diff
@@ -2382,6 +2386,58 @@ def test_remove_already_not_subbed(self) -> None:
         self.assert_length(json["removed"], 0)
         self.assert_length(json["not_removed"], 1)
 
+    def test_can_remove_subscribers_group(self) -> None:
+        realm = get_realm("zulip")
+        leadership_group = create_user_group(
+            "leadership", [self.example_user("iago"), self.example_user("shiva")], realm
+        )
+        managers_group = create_user_group("managers", [self.example_user("hamlet")], realm=realm)
+        add_subgroups_to_user_group(managers_group, [leadership_group])
+        cordelia = self.example_user("cordelia")
+
+        stream = self.make_stream("public_stream")
+
+        def check_unsubscribing_user(
+            user: UserProfile, can_remove_subscribers_group: UserGroup, allow_fail: bool = False
```

Comment:
I think we probably want to rename `allow_fail` to `expect_fail`, given we're actually asserting that it fails, not just being OK with it. I think the existing uses of `allow_fail` in the codebase are semantically incorrect too.